platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -756,6 +772,41 @@ def __install_firmware(self, image_path):
 
         return True
 
+    def __refresh_cpld(self, image_path):
+        with MPFAManager(image_path) as mpfa:
+            if not mpfa.get_metadata().has_option('firmware', 'refresh'):
+                raise RuntimeError("Failed to get {} refresh firmware".format(self.name))
+
+            refresh_firmware = mpfa.get_metadata().get('firmware', 'refresh')
+
+            print("INFO: Processing {} refresh file: firmware update".format(self.name))
+            self.__install_firmware(os.path.join(mpfa.get_path(), refresh_firmware))
+
+
+    def auto_update_firmware(self, image_path, boot_action):
+        """
+        Default handling of attempted automatic update for a component of a Mellanox switch.
+        Will skip the installation if the boot_action is 'warm' or 'fast' and will call update_firmware()
+        if boot_action is fast.
+        """
+
+        default_supported_boot = ['cold']
+
+        # Verify image path exists
+        if not os.path.exists(image_path):
+            # Invalid image path
+            return FW_AUTO_ERR_IMAGE
+
+        if boot_action in default_supported_boot:
+            # Defer refresh to next boot
+            if self.install_firmware(image_path):
+                return FW_AUTO_SCHEDULED
+            else:
+                return FW_AUTO_ERROR_UNKNOWN
+
+        # boot_type did not match (skip)
+        return FW_AUTO_ERR_BOOT_TYPE
```

Comment:
@alexrallen suggest reworking like this:
```python
        supported_boot = ['cold']

        # Verify image path exists
        if not os.path.exists(image_path):
            # Invalid image path
            return FW_AUTO_ERR_IMAGE

        if boot_action not in supported_boot:
            # boot_type did not match (skip)
            return FW_AUTO_ERR_BOOT_TYPE

        # Defer refresh to next boot
        if self.install_firmware(image_path):
            return FW_AUTO_SCHEDULED

        return FW_AUTO_ERROR_UNKNOWN
```

Reply:
Didn't end up with this exact code but I see the spirit of your comment I think. Updated these functions so they are syntactically similar to each other and easier to read / debug.